src/vs/workbench/browser/layout.ts

```diff
@@ -366,6 +392,49 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 		this.layout();
 	}
 
+	private setWindowBorder(skipLayout: boolean = false) {
+		if (isWeb || getTitleBarStyle(this.configurationService, this.environmentService) !== 'custom') {
+			return;
+		}
+
+		const theme = this.themeService.getTheme();
+
+		const activeBorder = theme.getColor(WINDOW_ACTIVE_BORDER);
+		const inactiveBorder = theme.getColor(WINDOW_INACTIVE_BORDER);
+
+		let windowBorder = false;
+		if (activeBorder || inactiveBorder) {
+			windowBorder = true;
+
+			let borderColor = this.state.hasFocus ? activeBorder : inactiveBorder;
+			if (!borderColor) {
```

Comment:
I do not really understand this fallback to the title color. When I tested this on mac I think I saw a white border around my window. Why do we need this?

Reply:
I added that fallback, because if there is only 1 color defined (active or inactive), then when the window is focused/unfocused the border would jump on and off.

Reply:
@eamodio I understand, but in that case wouldn't it be better to use the one color that is defined and not some random other color or even transparent?

Reply:
Yeah, the new code uses the 1 color that is defined